pkg/querier/vcs.go

```diff
@@ -0,0 +1,664 @@
+			)
+
+		}
+	})
+	return module, nil
+}
+
+func cleanWhiteSpace(s string) string {
+	space := false
+	return strings.Map(func(r rune) rune {
+		if r == '\n' || r == '\t' {
+			return -1
+		}
+		if r == ' ' && space {
+			return -1
+		}
+		space = r == ' '
+		return r
+	}, s)
+}
+
+// todo should move to golang package. and add tests
+func applyGoModule(modf *modfile.File, module goModuleFile) goModuleFile {
+	if modf == nil {
+		return module
+	}
+	for _, req := range modf.Require {
+		if req.Mod.Path == module.Path {
+			module.Version.Version = req.Mod.Version
+		}
+	}
+	for _, req := range modf.Replace {
+		if req.Old.Path == module.Path {
+			module.Path = req.New.Path
+			module.Version.Version = req.New.Version
+		}
+	}
+	return module
+}
+
+type goModuleFile struct {
+	module.Version
+	filePath string
+}
+
+func (m goModuleFile) String() string {
+	return fmt.Sprintf("%s@%s", m.Path, m.Version)
+}
+
+func parseGoModuleFilePath(path string) (goModuleFile, bool) {
+	parts := strings.Split(path, "@v")
+	if len(parts) != 2 {
+		return goModuleFile{}, false
+	}
+	first := strings.Index(parts[1], "/")
+	if first < 0 {
+		return goModuleFile{}, false
+	}
+	filePath := parts[1][first+1:]
+	modulePath := parts[0]
+	// searching for the first domain name
+	domainParts := strings.Split(modulePath, "/")
+	for i, part := range domainParts {
+		if strings.Contains(part, ".") {
+			return goModuleFile{
+				Version: module.Version{
+					Path:    strings.Join(domainParts[i:], "/"),
+					Version: "v" + parts[1][:first],
+				},
+				filePath: filePath,
+			}, true
+		}
+	}
+	return goModuleFile{}, false
+}
+
+func fetchGoMod(ctx context.Context, arg fileFinder) (*modfile.File, error) {
+	content, err := arg.client.GetFile(ctx, arg.repo.GetOwnerName(), arg.repo.GetRepoName(), "go.mod", arg.ref)
+	if err != nil {
+		return nil, err
+	}
+	return modfile.Parse("go.mod", []byte(content.Content), nil)
+}
+
+// findGoStdFile returns the file content if the given path is a standard go package.
+func findGoStdFile(ctx context.Context, arg fileFinder) (*vcsv1.GetFileResponse, error) {
+	if len(arg.path) == 0 {
+		return nil, nil
+	}
+	path := strings.TrimSuffix(arg.path, "/usr/local/go/src/")
+	path = strings.TrimSuffix(path, "$GOROOT/src/")
+	fileName := filepath.Base(arg.path)
+	packageName := strings.TrimSuffix(path, "/"+fileName)
+
+	// Todo: Send more metadata from SDK to fetch the correct version of Go std packages.
+	// For this we should use arbitrary k/v metadata in our request so that we don't need to change the API.
+	// I thought about using go.mod go version but it's a min and doesn't guarantee it hasn't been built with a higher version.
+	// Alternatively we could interpret the build system and use the version of the go compiler.
+	ref := "master"
+	isVendor := strings.HasPrefix(packageName, "vendor/")
+
+	if _, isStd := golang.StandardPackages[packageName]; !isVendor && !isStd {
+		return nil, nil
+	}
+	return fetchGoStd(ctx, path, ref)
```

Comment:
Probably can avoid a network call here in many cases if we download a few of the most popular go std lib versions to disk.

Reply:
Yeah that's faire but then we need to manage standard runtime version. Right now we default to latest because we can't figure which runtime version the binary has been built with but in the future we want that metadata to exists in pprof and use it to map the correct std go version.